Lib/pdb.py

```diff
@@ -755,6 +753,21 @@ def _complete_expression(self, text, line, begidx, endidx):
             # Complete a simple name.
             return [n for n in ns.keys() if n.startswith(text)]
 
+    def completedefault(self, text, line, begidx, endidx):
+        if text.startswith("$"):
```

Comment:
Maybe ``text[0] == '$'``?

Reply:
That has an implicit requirement that `text` is not empty - which I believe is not always true because when the user request a complete from nothing `text` would be empty. Also no matter if that condition holds or not, it's not obvious from the code and it requires more thinking.